### PR TITLE
Python: Expand `StringConstCompareBarrier` sanitizer gaurds to cover additional constants

### PR DESCRIPTION
**Pull request by joefarebrother**
Renames `StringConstCompareBarrier` to `ConstCompareBarrier` and expands it to cover comparisons to additional constats such as `None`. 
This reduces FP flow in cases such as `if x is None: return x`.

### CODE REVIEW

File: python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll [hunk 1]
```diff
@@ -142,7 +142,7 @@ module UrlRedirect {
   /**
    * A comparison with a constant string, considered as a sanitizer-guard.
    */
-  class StringConstCompareAsSanitizerGuard extends Sanitizer, StringConstCompareBarrier {
+  class StringConstCompareAsSanitizerGuard extends Sanitizer, ConstCompareBarrier {
```
Comment (yoff) [inline python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll:145]: Did you mean to follow the same pattern of deprecation here?

File: python/ql/test/library-tests/dataflow/tainttracking/commonSanitizer/test_const_compare.py [hunk 2]
```diff
@@ -85,6 +85,27 @@ def test_in_local_variable():
...
+def test_is_not_none():
+    ts = TAINTED_STRING
+    if ts is not None:
+        ensure_tainted(ts)  # $ tainted
+    else:
+        ensure_not_tainted(ts)
+
+def test_in_list_with_constants():
+    ts = TAINTED_STRING
+    if ts in ["safe", None, 3, False]:
```
Comment (yoff) [inline python/ql/test/library-tests/dataflow/tainttracking/commonSanitizer/test_const_compare.py:104]: Would it be considered tainted if there is a non-constant in this list? Should we have a test with that?

**Comment by yoff** [comment]
This looks like a very nice and useful generalisation :-)

File: python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll (hunk 1 above)
Comment (joefarebrother) [reply python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll]: Yes; I had missed it with a find-and-replace. 
Updated.

File: python/ql/test/library-tests/dataflow/tainttracking/commonSanitizer/test_const_compare.py (hunk 2 above)
Comment (joefarebrother) [reply python/ql/test/library-tests/dataflow/tainttracking/commonSanitizer/test_const_compare.py:104]: Additional test added.

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**